actors/test/miner_scenario_test.go

```diff
@@ -0,0 +1,315 @@
+
+	// assert successful precommit invocation
+	vm.ExpectInvocation{
+		// Original send to storage power actor
+		To:     minerAddrs.IDAddress,
+		Method: builtin.MethodsMiner.PreCommitSector,
+		Params: vm.ExpectObject(&preCommitParams),
+		SubInvocations: []vm.ExpectInvocation{
+			{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward},
+			{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower},
+			{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.VerifyDealsForActivation}},
+	}.Matches(t, v.Invocations()[0])
+
+	// find information about precommited sector
+	var minerState miner.State
+	err = v.GetState(minerAddrs.IDAddress, &minerState)
+	require.NoError(t, err)
+
+	precommit, found, err := minerState.GetPrecommittedSector(v.Store(), sectorNumber)
+	require.NoError(t, err)
+	require.True(t, found)
+
+	// advance time to max seal duration
+	proveTime := v.GetEpoch() + miner.MaxSealDuration[sealProof]
+	v, dlInfo := vm.AdvanceTillEpoch(t, v, minerAddrs.IDAddress, proveTime)
+
+	//
+	// overdue precommit
+	//
+
+	t.Run("missed prove commit results in precommit expiry", func(t *testing.T) {
+		// advanced one more deadline so precommit is late
+		tv, _ := v.WithEpoch(dlInfo.Close)
+		require.NoError(t, err)
+
+		// run cron which should expire precommit
+		_, code = tv.ApplyMessage(builtin.SystemActorAddr, builtin.CronActorAddr, big.Zero(), builtin.MethodsCron.EpochTick, nil)
+		require.Equal(t, exitcode.Ok, code)
+
+		vm.ExpectInvocation{
+			// Original send to storage power actor
+			To:     builtin.CronActorAddr,
+			Method: builtin.MethodsCron.EpochTick,
+			SubInvocations: []vm.ExpectInvocation{
+				{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.OnEpochTickEnd, SubInvocations: []vm.ExpectInvocation{
+					{To: minerAddrs.IDAddress, Method: builtin.MethodsMiner.OnDeferredCronEvent, SubInvocations: []vm.ExpectInvocation{
+						{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward},
+						{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower},
+
+						// The call to burnt funds indicates the overdue precommit has been penalized
+						{To: builtin.BurntFundsActorAddr, Method: builtin.MethodSend, Value: vm.ExpectAttoFil(precommit.PreCommitDeposit)},
+						{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.EnrollCronEvent},
+					}},
+					//{To: minerAddrs.IDAddress, Method: builtin.MethodsMiner.ConfirmSectorProofsValid},
+					{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.UpdateNetworkKPI},
+				}},
+				{To: builtin.StorageMarketActorAddr, Method: builtin.MethodsMarket.CronTick},
+			},
+		}.Matches(t, tv.Invocations()[0])
+	})
+
+	//
+	// prove and verify
+	//
+
+	// Prove commit sector after max seal duration
+	v, err = v.WithEpoch(proveTime)
+	require.NoError(t, err)
+	proveCommitParams := miner.ProveCommitSectorParams{
+		SectorNumber: sectorNumber,
+	}
```

Comment:
Why does `Proof == nil` work here?

Reply:
The miner_actor simply passes the `Proof` into it's verify syscall and the scenario test VM syscall simply ignores the input and does nothing indicating the proof is valid. Scenario testing would get extremely complicated if we needed real proofs, and the unit tests cover that the miner handles the proof correctly.

Reply:
Makes sense. Do you think it'd be useful to at least check that syscalls are made? Something like `ExpectInvocation`, but with important syscalls.

Reply:
It would be easy to do. Those syscalls are well covered in the unit tests, so I haven't bothered here. I'm going to state that things that we can test well in unit tests we should avoid in scenario tests just to keep complexity down and reduce the burden of change. I could easily be convinced otherwise if you or someone else were to make the case.